core/README.md

```diff
@@ -0,0 +1,160 @@
+
+When you want to get the generated data, you would **build** the model (builder).
+
+Builders are essentially composed by a generator and (optionally) transformers.
+
+### Generator
+
+A generator is what describes the initial model shape. Most of the fields can and should be initialized with some values. You can define random values using `fake` or static values.
+
+```ts
+const generator = Generator<TAuthor>({
+  name: 'Author',
+  fields: {
+    firstName: fake((f) => f.name.firstName()),
+    locale: 'en',
+  },
+});
+```
+
+> The shape of the generator model is what usually is used in the UI and does not need transformation.
+
+Models can have nested fields, like object or arrays. Most of the time those are also defined as models.
+In this case, the generator can define the value of the nested field as `null`. Those nested models are then built in the builder and or the transformers.
+
+For example, the author model can have a list of books:
+
+```ts
+const generator = Generator<TAuthor>({
+  name: 'Author',
+  fields: {
+    firstName: fake((f) => f.name.firstName()),
+    locale: 'en',
+    books: null,
+  },
+});
+```
+
+Books are defined as another model:
+
+```ts
+const generator = Generator<TBook>({
+  name: 'Book',
+  fields: {
+    title: fake((f) => f.random.words()),
+  },
+});
+```
+
+When defining the builder for the `Author`, we can then initialize the `books` field by referencing the `Book` model.
+
+```ts
+const Author = Builder<TAuthor>({
+  generator,
+  transformers,
+}).books([Book.random()]);
+```
+
+At the same time, we also need to tell the transformers that the `books` field can be built. We do that by specifying the field in the `buildFields` option:
+
+```ts
+const transformers = {
+  default: Transformer<TAuthor, TAuthor>('default', {
+    buildFields: ['books'],
+  }),
+  graphql: Transformer<TAuthor, TAuthorGraphql>('graphql', {
+    buildFields: ['books'],
+    addFields: () => ({
+      __typename: 'Author',
+    }),
+  }),
+};
+```
+
+### Transformers
+
+A transformer is used to transform the data defined in the generator (the initial model) to another shape.
+
+There are 3 available transformer types:
+
+- `default`: this is the "identity" transformer. You would use the `build()` method to get the generated data according to the initial shape.
+- `graphql`: this is the transformer for GraphQL data shapes. You would use the `buildGraphql()` method to get the generated data transformed into a GraphQL shape.
+- `rest`: this is the transformer for REST data shapes. You would use the `buildRest()` method to get the generated data transformed into a REST shape.
+
+Defining a `graphql` and/or a `rest` transformer is optional and depends on what the specific data requirements are.
+
+For example, for the `Author` we can define a `graphql` transformer like this:
+
+```ts
+const transformers = {
+  default: Transformer<TAuthor, TAuthor>('default', {
+    buildFields: ['books'],
+  }),
```

Comment:
Irrelevant.

```suggestion
```

Reply:
Again, why is it irrelevant? 🤔

Reply:
The emphasis understood from the following sentence is the GraphQL transformer. Adding other stuff can make the reader distracted, in my opinion.

> For example, for the `Author` we can define a `graphql` transformer like this:

Reply:
Yep ok 👌